packages/@glimmer/integration-tests/test/helpers/hash-test.ts

```diff
@@ -193,13 +193,36 @@ class HashTest extends RenderTest {
 
     this.registerComponent('Glimmer', 'FooBar', `{{yield @hash}}`, FooBar);
 
-    assert.throws(() => {
-      this.render(`
-        <FooBar @hash={{hash firstName="Godfrey"}} as |values|>
-          {{values.firstName}} {{values.lastName}}
-        </FooBar>
-      `);
-    }, /You attempted to set the "firstName" value on an object generated using the \(hash\) helper|Assignment to read-only properties is not allowed in strict mode/);
+    this.render(
+      `<FooBar @hash={{hash firstName="Godfrey" lastName="Hietala"}} as |values|>{{values.firstName}} {{values.lastName}}</FooBar>`
+    );
+
+    this.assertHTML('Chad Hietala');
+    this.assertStableRerender();
+  }
+
+  @test
+  'defined hash keys are reset whenever the upstream changes'() {
```

Comment:
I think the prior behavior was that all manually set hash values would be reset if *any* were reset (because the hash itself was unstable), right?

Could you add another test for that?

Reply:
We cannot do this, unfortunately. This directly contradicts with our goal of making each individual key lazy. The way that was accomplished previously was the fact that the value of a reference was different than the paths of references in templates.

Basically this would put us back to square one.

Reply:
Discussed it further with @rwjblue, what we can do is eagerly entangle anything the first time a value is set on the hash, and then reset everything whenever anything changes. It's a bit annoying, but it's more accurate compared to the original behavior.

Reply:
Discussed further again, that strategy unfortunately does not work in IE11, so we're back to the original new semantics